src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullChecking.cs

```diff
@@ -10,9 +10,41 @@ namespace Microsoft.CodeAnalysis.CSharp
     internal sealed partial class LocalRewriter
     {
         private BoundStatement RewriteNullChecking(BoundBlock block)
+        {
+            var statementList = ConstructNullCheckedStatementList(_factory.CurrentFunction.Parameters, block.Statements);
+            if (statementList.IsDefaultOrEmpty)
+            {
+                return block;
+            }
+            return block.Update(statementList);
+        }
+
+        private BoundLambda RewriteNullChecking(BoundLambda lambda)
+        {
+            var statementList = ConstructNullCheckedStatementList(_factory.CurrentFunction.Parameters, lambda.Body.Statements);
+            if (statementList.IsDefaultOrEmpty)
```

Comment:
I don't think `IsDefaultOrEmpty` is appropriate here. We don't return both, so if we decide that the sentinel value is `default` we should check `IsDefault` and vice versa if it's Empty. In fact, we should probably Debug.Assert that it's not the other one, since that would be a bug.

Reply:
I changed to match the new behavior on returning the statement List directly, can swap back.

Reply:
FIxed based on in-person discussion